pkg/json/ingest.go

```diff
@@ -118,7 +121,12 @@ func (ingester *Ingester) ingestPolymorphicNode(input jsonom.Node, path ls.NodeP
 		if err != nil {
 			return nil, err
 		}
-		dp = x
+		if n == nil {
+			return nil, ls.ErrSchemaValidation{Msg: "invalid schema validation", Path: path}
```

Comment:
Remove this section completely, polymorphic will not be called if schemaNode is nil